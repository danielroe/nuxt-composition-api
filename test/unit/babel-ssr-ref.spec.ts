/* eslint-disable @typescript-eslint/no-var-requires */
const babel = require('@babel/core')
const plugin = require('../../lib/babel')
/* eslint-enable */

const example = `
const ref = ref(1)
const ref2 = ssrRef(2)
const ref3 = ssrRef(3, 'custom-key')
const ref4 = ssrRef(() => 4)

const async1 = useAsync(() => null)
const async2 = useAsync(() => null, 'key')

const stat = useStatic(() => 4, '2', 'post')
const stat2 = useStatic(() => 4, '2')
const stat3 = useStatic(() => 4)
`

describe('ssrRef babel plugin', () => {
  it('works', () => {
    const { code } = babel.transform(example, { plugins: [plugin] })!
    expect(code).toMatchSnapshot()
  })
})
