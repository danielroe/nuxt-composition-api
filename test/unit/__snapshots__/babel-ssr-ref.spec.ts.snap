// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssrRef babel plugin works 1`] = `
"const ref = ref(1);
const ref2 = ssrRef(2, \\"ref2-BAW7YFDj4E+Qxr+ujqEADg==\\");
const ref3 = ssrRef(3, 'custom-key');
const ref4 = ssrRef(() => 4, \\"ref4-h6IKM1doqCRBR49lWv2V/g==\\");
const async1 = useAsync(() => null, \\"async1-nk1uJ/q39HMoKEMFH33Ryg==\\");
const async2 = useAsync(() => null, 'key');"
`;
